Jasmine/Tetris/ViewModels/TetrisGameViewModel.swift

```diff
@@ -1,4 +1,158 @@
 import Foundation
 
+/// The main view model for Tetris game
 class TetrisGameViewModel {
+
+    weak var delegate: TetrisGameViewControllerDelegate?
+
+    fileprivate var tetrisGrid = TetrisGrid()
+
+    var upcomingTiles: [String] = []
+    fileprivate var fallingTileText: String?
+
+    var currentScore: Int = 0
+
+    let countDownTimer = CountDownTimer(totalTimeAllowed: Constants.Tetris.totalTime)
+
+    /// Populate upcomingTiles
+    init() {
+        for _ in 0..<Constants.Tetris.upcomingTilesCount {
+            upcomingTiles.append(getRandomWord())
+        }
+    }
+
+    /// Checks for and returns coordinates of matching phrase, searching by row-wise and return if found
+    /// otherwise continue with searching by column-wise
+    fileprivate func checkForMatchingPhrase() -> Set<Coordinate>? {
+        if let matchedCoordinates = checkForMatchingPhrase(searchByRow: true) {
```

Comment:
it's quite weird to refactor the method for it to be used for both searching by row and by column but that's the only way to ensure DRY